pennylane_lightning_gpu/src/algorithms/ObservablesGPU.hpp

```diff
@@ -562,7 +562,7 @@ class SparseHamiltonianGPU final : public ObservableGPU<T> {
             /* cusparseDnVecDescr_t */ sv_prime_descriptor,
             /* cudaDataType */ data_type,
             /* cusparseSpMVAlg_t */
-            CUSPARSE_MV_ALG_DEFAULT,
+            CUSPARSE_SPMV_ALG_DEFAULT,
```

Comment:
```suggestion
            CUSPARSE_SPMV_CSR_ALG1,
```

Let's try the above, as it mirrors what is at line 548.